typing/typecore.ml

```diff
@@ -5702,12 +5767,18 @@ and map_half_typed_cases
     end
   in
   (* type bodies *)
+  let half_typed_cases_cont_list =
+    match conts with
+    | None -> List.map (fun x -> (x, None)) half_typed_cases
+    | Some conts ->
+        List.map2 (fun x cont -> (x, cont)) half_typed_cases conts
```

Comment:
This works only because effect cases (which have one continuation each) are typed separately than other cases, so the lists have the same size. It might be useful to explain this invariant here, or to move this mapping outside of [map_half_typed_cases], close to the call-sites of the function where it is clear that the lists are the same size ?